### ISSUE TITLE
Feature: peer reviews for advanced challenges

### ISSUE THREAD

**Issue by marzelin** [freeCodeCamp/freeCodeCamp#11865]
# Feature: peer reviews for advanced challenges

##  The problem

Since writing code is becoming more and more collaborative process the ability to produce highly readable code is becoming crucial for aspiring developers. Writing code that is easy to understand by other people is as important as writing code that fulfils technical requirements.
There are tools that can check and enforce some aspects of writing clear code, but they are not able to check all the things, specifically the tools cannot check if:

- the names for identifiers are appropriate
- quality of documentation/comments
- overall structure of code that matters for human comprehension and is completely ambivalent for machines

Also, ability to read code with ease is also important and should be practiced as part of the course.

## Proposed solution: peer reviews

There should be a peer review system for non-trivial challenges.
After finishing a non-trivial challenge, a user should review three solutions submitted by other students before continuing to the next challenge.

### Examples of existing platforms that use peer reviews:
- EdX platform
- Coursera
### Drawbacks:
- since creating peer-review system is a quite complex task it requires a substantial amount of time
- peer reviews are not always accurate since they are written by students not teachers

**Comment by erictleung** [comment]
@marzelin thanks for the feature suggestion. I think this would be an interesting twist to the program that would be similar to the kind of programming you might do on the job. Another drawback, similar to your second point, is that peer review is subjective on what would be considered good. And to clarify, which advanced challenges are you referring to? All of them? Only adv. algorithms? The projects?

For now, there is a Gitter room for code review https://gitter.im/FreeCodeCamp/CodeReview. That might be the easiest to get feedback if this feature is not implemented.

**Comment by dhcodes** [comment]
I like this idea, but recommend the following:
a) guidelines be set up to help promote good reviews (to combat @erictleung's point about quality of feedback)
b) it be optional. It could be rewarded with a brownie point(s) like some other aspects of the site. 

Soon, the projects will implement testing to ensure the user stories are met. That will help to ensure functionality. I think code quality is also important, but one problem might be agreeing on what constitutes quality code especially among campers who may lack experience.

tldr; I like it but am not sure how to best implement it.
